Tests/SubscriptionTests/Managers/SubscriptionManagerV2Tests.swift

```diff
@@ -28,20 +28,29 @@ class SubscriptionManagerV2Tests: XCTestCase {
     var mockOAuthClient: MockOAuthClient!
     var mockSubscriptionEndpointService: SubscriptionEndpointServiceMockV2!
     var mockStorePurchaseManager: StorePurchaseManagerMockV2!
+    var mockAppStoreRestoreFlowV2: AppStoreRestoreFlowMockV2!
+    var overrideTokenResponse: Result<Networking.TokenContainer, Error>?
 
     override func setUp() {
         super.setUp()
 
         mockOAuthClient = MockOAuthClient()
         mockSubscriptionEndpointService = SubscriptionEndpointServiceMockV2()
         mockStorePurchaseManager = StorePurchaseManagerMockV2()
+        mockAppStoreRestoreFlowV2 = AppStoreRestoreFlowMockV2()
 
         subscriptionManager = DefaultSubscriptionManagerV2(
             storePurchaseManager: mockStorePurchaseManager,
             oAuthClient: mockOAuthClient,
             subscriptionEndpointService: mockSubscriptionEndpointService,
             subscriptionEnvironment: SubscriptionEnvironment(serviceEnvironment: .staging, purchasePlatform: .stripe),
-            pixelHandler: { _ in }
+            pixelHandler: { _ in },
+            autoRecoveryHandler: {
```

Comment:
this is an example of the new handler implementation, ignore the `overrideTokenResponse ` that is used only for testing purposes